packages/netlify-cms-core/src/reducers/entries.ts

```diff
@@ -643,7 +643,7 @@ function evaluateFolder(
   entryMap: EntryMap | undefined,
   field: EntryField | undefined,
 ) {
-  let currentFolder = config[folderKey];
+  let currentFolder = config[folderKey]!;
```

Comment:
I'm not sure this is a correct way to handle it. Because here we're lying to compiler.

Reply:
I'm wondering if there is more correct way to express it with types, so we don't have to use these escape hatches.